src/suggestion.rs

```diff
@@ -135,6 +158,131 @@ impl<'s> fmt::Display for Suggestion<'s> {
+        //    |                                                                                                                                                                                                ^^^
+        //    | - not, Ont, int, on, onto, cont, font, or one of 9 others
+        //    |
+        //    |   Possible spelling mistake found.
+        //    |
+        // ```
+        const PADDING_OFFSET: usize = 6;
+        const PADDING_OFFSET_START: usize = 4;
+        const PADDING_END: usize = 15;
+        const TOO_LONG_WORD: usize = 20;
+        const DISPLAYED_LONG_WORD: usize = 4usize;
+
+        let terminal_size: usize = get_terminal_size();
+        // We will be using ranges to help doing the fitting:
+        //
+        // |----------------------------------literal_word----------------------------------|
+        // |----------------------|---------misspelled_word---------|-----------------------|
+        // |-----left_context-----|---start_word----|----end_word---|-----right_context-----|
+        //
+        let mut range_left_context = Range {
+            start: 0usize,
+            end: marker_range_relative.start,
+        };
+        let mut range_right_context = Range {
+            start: marker_range_relative.end,
+            end: self.literal.as_str().chars().count(),
+        };
+        let mut range_start_word = Range {
+            start: marker_range_relative.start,
+            end: marker_range_relative.start,
+        };
+        let mut range_end_word = Range {
+            start: marker_range_relative.end,
+            end: marker_range_relative.end,
+        };
+        if self.literal.as_str().chars().count() > terminal_size {
+            let mut misspelled_word = format!(
+                "{}",
+                self.literal.sub(Range {
+                    start: marker_range_relative.start - 1,
+                    end: marker_range_relative.end
+                })
+            );
+            if marker_size > TOO_LONG_WORD {
+                range_start_word.start = marker_range_relative.start - 1;
+                range_start_word.end = range_start_word.start + DISPLAYED_LONG_WORD;
+
+                range_end_word.start = marker_range_relative
+                    .end
+                    .saturating_sub(DISPLAYED_LONG_WORD);
+                range_end_word.end = marker_range_relative.end;
+
+                misspelled_word = format!(
+                    "{}...{}",
+                    self.literal.sub(range_start_word),
+                    self.literal.sub(range_end_word)
+                );
+                marker_size = misspelled_word.chars().count();
+            };
+            // right context has enough info to fill the terminal
+            //  |-----misspelled_word-----|--------right_context---------|
+            //
+            // Attempt to fit the misspelled word in the beginning followed by info.
+            if range_right_context.len() >= terminal_size {
```

Comment:
```suggestion
            if range_right_context.len() >= terminal_size + marker_size  {
```
I think for this check you need to take into account the actual mistake on top of the context, otherwise we'd get inconsistent behaviour depending on the length of the word that is misspelled

Reply:
not really, if I would be more precise I would also consider this `padding_till_literal_start` variable. 

But my goal here is just to have more info than needed so I can use it to fill the terminal.

Reply:
`marker_size` is not included in the range_right_context so I do not need to exclude it. This is not super precise as I would only need `terminal_size - padding_till_literal_start - any_extra_formatting`, but it was just to have a rough idea that I have enough context to fill the terminal.

Reply:
but the goal should be to make sure the marked string, and left as well as right minimum context fit into a line, so the comparison with `terminal_size` here seems incorrect to me. Implementing unit test might be helpful here :)